### PR TITLE
[New Extension] Podcasts Now

### PR DESCRIPTION
**Pull request by cloudy9101**
## Description

Podcasts Now is an extension that allows users to manage, and listen to their podcasts through Raycast based on Music.app.

## Screencast

podcasts-now-4
podcasts-now-3
podcasts-now-1
podcasts-now-2

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/podcasts-now/src/add-podcast.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      }
+    });
+    await LocalStorage.setItem(PODCASTS_FEEDS_KEY, JSON.stringify(uniq(validUrls)));
+    popToRoot({ clearSearchBar: false });
+  };
+
+  return (
+    <Form
+      actions={
+        <ActionPanel title="#1 in raycast/extensions">
```
Comment (thomaslombart) [inline extensions/podcasts-now/src/add-podcast.tsx:60]: I'm not sure I get what does the title `"#1 in raycast/extensions"` mean here?

File: extensions/podcasts-now/src/hooks/usePodcasts.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { LocalStorage, Cache } from "@raycast/api";
+import { useEffect, useState } from "react";
+import { forEach } from "lodash";
+
+import { PODCASTS_FEEDS_KEY } from "../constants";
+import { getFeed, Podcast } from "../feed-parser";
+
+const cache = new Cache({ namespace: "podcasts" });
+
+export const usePodcasts = () => {
```
Comment (thomaslombart) [inline extensions/podcasts-now/src/hooks/usePodcasts.ts:10]: Same thing here, I think `useCachedPromise` will probably simplify things. 😊

File: extensions/podcasts-now/src/podcast-list.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        />
+      </List.Section>
+      <List.Section title="Podcasts">
+        {filteredList.map((item) => {
+          return (
+            <List.Item
+              id={item.feedUrl}
+              key={item.feedUrl}
+              icon={{
+                source: item?.itunes?.image || "",
```
Comment (thomaslombart) [inline extensions/podcasts-now/src/podcast-list.tsx:110]: Could you return a default icon to show? It looks empty on my end.

podcasts-now 2023-02-08 at 11 11 59

**Review by thomaslombart — CHANGES_REQUESTED** [review]
Thanks for this great extension 🎧

I've now reviewed it, and I have a couple of improvements to suggest:

- Some actions don't have any shortcuts assigned (like Rewinding the playing episode), could you add them? Some folks are heavy keyboard users, so this would be appreciated.
- It'd be awesome to add a Menu Bar command to this extension. This command could mimic the list structure. Menu Bar commands are perfect for "live" extensions like this one. (this isn't required, though! Don't add that now if you don't have time 😄)
-  I have some flickering when playing/pausing the current episode. Could you take a look and maybe improve that? Ideally, pausing and playing should trigger the loading state once, and the podcast icons shouldn't flicker.

https://user-images.githubusercontent.com/16003285/217501394-d67ecccb-0bfe-4d7c-b712-8aef3a38e20b.mp4

File: extensions/podcasts-now/src/add-podcast.tsx (hunk 1 above)
Comment (pernielsentikaer) [reply extensions/podcasts-now/src/add-podcast.tsx:60]: This is the default title from one of our examples 😊

Comment (thomaslombart) [reply extensions/podcasts-now/src/add-podcast.tsx:60]: But that doesn't have anything to do with this extension, right? 😊

Comment (pernielsentikaer) [reply extensions/podcasts-now/src/add-podcast.tsx:60]: No, just a "leftover" 😄

File: extensions/podcasts-now/src/podcast-list.tsx (hunk 3 above)
Comment (cloudy9101) [reply extensions/podcasts-now/src/podcast-list.tsx]: Add a default icon through the `fallback` attr

**Comment by cloudy9101** [comment]
@thomaslombart Thank you for reviewing. Your suggestions are amazing which lead me to do a bit refactor and make the extension have better performance and user experience. I also appreciate you found all the grammar mistakes in README and package.json files. As not a native English speaker, I should take them to go through Grammarly first.

I refactor the code a bit and also implement the menu bar command which it's totally a great feature for me and this extension. Please review it again.

**Comment by thomaslombart** [comment]
Hey @cloudy9101, sorry for the late review. I was on vacation. I tried testing your extension again but couldn't because of an error:

podcasts-now 2023-02-21 at 09 56 09

Could you fix it so I can test the rest of the extension? 😊

**Comment by cloudy9101** [comment]
Hi @thomaslombart , the error should be fixed, could you please test again?

**Review by thomaslombart — APPROVED** [review]

**Pull request merged**